dio/lib/src/dio_event_processor.dart

```diff
@@ -89,6 +89,24 @@ class DioEventProcessor implements EventProcessor {
       headers: _options.sendDefaultPii ? headers : null,
       bodySize: dioError.response?.data?.length as int?,
       statusCode: response?.statusCode,
+      data: _getResponseData(dioError.response?.data),
     );
   }
+
+  /// Returns the response data, if possible according to the users settings.
+  Object? _getResponseData(dynamic data) {
```

Comment:
Lets stick to `Object` instead of `dynamic` so we can make it nullable or not.

Reply:
The`data` we get from the dio exception is marked as `dynamic`, it's the same API as used for the request data.

Reply:
Yep, we can refactor both to use `Object`.
We had this issue at some point https://github.com/getsentry/sentry-dart/issues/511 but it was just too much work for not a lot of benefit, we can though use `Object` or `Object?` for new code, so we can use the null safety feature during compilation time.